api/externalLinks/ExternalLinksRestHandler.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		impl.logger.Errorw("service err, SaveLink", "err", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+func (impl ExternalLinksRestHandlerImpl) GetExternalLinksTools(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+
+	// auth free api as we using this for multiple places
+	/*
+		token := r.Header.Get("token")
+		if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionGet, "*"); !ok {
+			common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+			return
+		}
+	*/
+
+	res, err := impl.externalLinksService.GetAllActiveTools()
+	if err != nil {
+		impl.logger.Errorw("service err, GetAllActiveTools", "err", err)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+func (impl ExternalLinksRestHandlerImpl) GetExternalLinks(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	v := r.URL.Query()
+	id := v.Get("clusterId")
+	clusterId := 0
+	if len(id) > 0 {
+		clusterId, err = strconv.Atoi(id)
+		if err != nil {
+			impl.logger.Errorw("request err, FetchOne", "err", err, "id", id)
+			common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+			return
+		}
+	}
+
+	//apply auth only in case when requested for all links
+	if clusterId == 0 {
+		token := r.Header.Get("token")
+		if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionGet, "*"); !ok {
+			common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+			return
+		}
+	}
+
+	res, err := impl.externalLinksService.FetchAllActiveLinks(clusterId)
+	if err != nil {
+		impl.logger.Errorw("service err, FetchAllActive", "err", err)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+func (impl ExternalLinksRestHandlerImpl) UpdateExternalLinks(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	var bean externalLinks.ExternalLinksRequest
+	err = decoder.Decode(&bean)
+	if err != nil {
+		impl.logger.Errorw("request err, Update Link", "err", err, "bean", bean)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	bean.UserId = userId
+
+	token := r.Header.Get("token")
+	if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionUpdate, "*"); !ok {
```

Comment:
Is this superadmin check? please confirm

Reply:
yes its a global resource, its general check